indico/modules/events/sessions/ical.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from icalendar.cal import Calendar
+from werkzeug.urls import url_parse
+
+from indico.core.config import config
+from indico.modules.events.ical import generate_basic_component
+from indico.web.flask.util import url_for
+
+
+def generate_session_component(session, related_to_uid=None):
+    """Generates an Event icalendar component from an Indico Session.
+
+    :param session: The Indico Session to use
+    :param related_to_uid: Indico uid used in related_to field
+    :returns: an icalendar Event
+    """
+
+    uid = 'indico-session-{}@{}'.format(session.id, url_parse(config.BASE_URL).host)
```

Comment:
Could use an f-string here:

```suggestion
    uid = f'indico-session-{session.id}@{url_parse(config.BASE_URL).host}'
```